frame/name-service/src/lib.rs

```diff
@@ -0,0 +1,480 @@
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		// TODO: Potentially make events more lightweight
+		/// A new `Commitment` has taken place.
+		Committed { sender: T::AccountId, who: T::AccountId, hash: CommitmentHash },
+		/// A new `Registration` has taken added.
+		NameRegistered { name_hash: NameHash, owner: T::AccountId, expiry: T::BlockNumber },
+		/// A `Registration` has been transferred to a new owner.
+		Transfer { from: T::AccountId, to: T::AccountId },
+		/// A `Registration` has been renewed.
+		NameRenewed { name_hash: NameHash, expires: T::BlockNumber },
+		/// An address has been set for a name hash to resolve to.
+		AddressSet { name_hash: NameHash, address: T::AccountId },
+		/// An address was deregistered.
+		AddressDeregistered { name_hash: NameHash },
+	}
+
+	// Your Pallet's error messages.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// This commitment hash already exists in storage.
+		AlreadyCommitted,
+		/// It has not passed the minimum waiting period to reveal a commitement.
+		TooEarlyToReveal,
+		/// The commitment cannot yet be removed. Has not expired.
+		CommitmentNotExpired,
+		/// The commitment has expired.
+		CommitmentExpired,
+		/// This commitment does not exist.
+		CommitmentNotFound,
+		/// This name is already registered.
+		AlreadyRegistered,
+		/// This registration does not exist.
+		RegistrationNotFound,
+		/// The sender is not the registration owner.
+		NotRegistrationOwner,
+		/// This resolver does not exist.
+		ResolverNotFound,
+		/// This registration has expired.
+		RegistrationExpired,
+		/// This registration has not yet expired.
+		RegistrationNotExpired,
+		/// Conversion error
+		ConversionError,
+	}
+
+	// Your Pallet's callable functions.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		// TODO: Should we allow registration on behalf of?
+		#[pallet::weight(0)]
+		pub fn commit(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			commitment_hash: CommitmentHash,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			ensure!(!Commitments::<T>::contains_key(commitment_hash), Error::<T>::AlreadyCommitted);
+
+			let block_number = frame_system::Pallet::<T>::block_number();
+			let deposit = T::CommitmentDeposit::get();
+
+			T::Currency::reserve(&sender, deposit)?;
+
+			let commitment = Commitment { who: who.clone(), when: block_number, deposit };
+
+			Commitments::<T>::insert(commitment_hash, commitment);
+			Self::deposit_event(Event::<T>::Committed { sender, who, hash: commitment_hash });
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn reveal(
+			origin: OriginFor<T>,
+			name: Vec<u8>,
+			secret: u64,
+			periods: u32,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			let commitment_hash = sp_io::hashing::blake2_256(&(&name, secret).encode());
+
+			let commitment =
+				Commitments::<T>::get(commitment_hash).ok_or(Error::<T>::CommitmentNotFound)?;
```

Comment:
```suggestion
				Commitments::<T>::take(commitment_hash).ok_or(Error::<T>::CommitmentNotFound)?;
```

And no need to remove later